packages/hardhat-zksync-solc/src/compile/downloader.ts

```diff
@@ -235,5 +259,9 @@ export class ZksolcCompilerDownloader {
         if (versionOutput.status !== 0 || version === null) {
             throw new ZkSyncSolcPluginError(COMPILER_BINARY_CORRUPTION_ERROR(compilerPath));
         }
+
+        if (this._configCompilerPath) {
+            this._version = version!;
+        }
```

Comment:
Does it make sense also to add check that if someone set version in hh config, to compare it with the version obtained by calling compiler? In that way we can prohibit if the versions is not matching. Also, than there is new role: if you want to try some other release that is not in the range of allowed versions from version info file, than don't specify any version.

Reply:
Error handling is supported differently than suggested; thus, users need to specify 'remote' as the version or leave it empty if they intend to use the remote origin.